bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java

```diff
@@ -211,13 +212,21 @@ public void initialize() {
                 return UpdatesListener.CONFIRMED_UPDATES_ALL;
         }, exception -> {
                 if (exception != null) {
-                    logger.warn("Telegram exception: {}", exception.getMessage());
+                    logger.debug("Telegram exception: {}", exception.getMessage());
                     if (exception.response() != null) {
                         BaseResponse localResponse = exception.response();
-                        if (localResponse.errorCode() == 401) {
-                            logger.error("Bot token invalid, disable thing {}", getThing().getUID());
-                            localBot.removeGetUpdatesListener();
-                            updateStatus(ThingStatus.OFFLINE);
+                        switch(localResponse.errorCode()) {
+                            case 401: {
+                                logger.error("Bot token invalid, disable thing {}", getThing().getUID());
+                                localBot.removeGetUpdatesListener();
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "Bot token invalid.");
+                                break;
+                            }
+                            default: {
+                                logger.error("Unknown error {}", getThing().getUID());
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "Unknown error " + localResponse.errorCode());

```

Comment:
I'm not sure if it's a good idea to immediately set the thing to OFFLINE for any error code. Some error codes might be related to a faulty API call e.g. the user want's to send an image which is too big which results in a error "400 BAD_REQUEST" with PHOTO_INVALID_DIMENSIONS. But this doesn't mean that the overall connection is broken. So you could still send other messages etc.

Reply:
Thanks for your response. I also thought about that. My first idea that will be no default case, only the UnknownHostException and the Bot token will be handled here, but I haven't found any reference that what error code is returned by the TelegramAPI when UnknownHostError happens...

Reply:
UnknowHostException is a java exception. You could check which type of exception you get with `instanceof` and distinguish between protocol exceptions and other exceptions.